### PR TITLE
[Test] Apply UnitTest for UZService

### PR DESCRIPTION
**Pull request by namnh-0652**
**Description**
- Apply UnitTest for UZService

**Impacted Area**
- None

**Ticket URLs**
- https://uizaio.atlassian.net/browse/EBC-435
- https://uizaio.atlassian.net/browse/EBC-436

### CODE REVIEW

**Comment by namnh-0652** [comment]
@deathemperor This PR is not finished yet. So, please don't merge ! 
Thanks !

**Comment by namnh-0652** [comment]
@deathemperor Please help to recheck it again. The fully report is below:

No | Method  | Is using in SDK or Sample  | Testing Status |
| :--------------- | :------------ |:---------------:| -----:|
| 1 |  POST("/api/public/{api_version}/admin/user/auth") **getToken()**    | &#9744; | &#9744; |
| 2 |  POST("/api/public/{api_version}/admin/user/auth/check-token") **checkToken()**      | &#9744; | &#9744; |
| 3 |  POST("/api/public/{api_version}/admin/user") **createAnUser()**    | &#9745; | &#9744; |
| 4 |  GET("/api/public/{api_version}/admin/user") **retrieveAnUser()**      | &#9745; | &#9745; |
| 5 |  GET("/api/public/{api_version}/admin/user") **listAllUser()**    | &#9745; | &#9744; |
| 6 |  PUT("/api/public/{api_version}/admin/user") **updateAnUser()**      | &#9745; | &#9745; |
| 7 |  HTTP(method = "DELETE", path = "/api/public/v4/admin/user", hasBody = true) **deleteAnUser()**    | &#9745; | &#9744; |
| 8 |  PUT("/api/public/{api_version}/admin/user/changepassword") **updatePassword()**      | &#9745; | &#9744; |
| 9 |  GET("/api/public/{api_version}/media/metadata") **getListMetadata()**    | &#9745; | &#9745; |
| 10 | GET("/api/public/{api_version}/media/metadata") (paging) **getListMetadata()**      | &#9745; | &#9745; |
| 11 |  POST("/api/public/{api_version}/media/metadata") **createMetadata()**    | &#9745; | &#9745; |
| 12 |  GET("/api/public/{api_version}/media/metadata") **getDetailOfMetadata()**      | &#9745; | &#9745; |
| 13 |  PUT("/api/public/{api_version}/media/metadata") **updateMetadata()**    | &#9745; | &#9745; |
| 14 |  DELETE("/api/public/{api_version}/media/metadata") **deleteAnMetadata()**      | &#9745; | &#9745; |
| 15 |  GET("/api/public/{api_version}/media/entity") **getListAllEntity()**    | &#9745; | &#9745; |
| 16 |  GET("/api/public/{api_version}/media/entity") **getListAllEntity()**      | &#9745; | &#9745; |
| 17 |  GET("/api/public/{api_version}/media/entity") **getListAllEntity()**    | &#9744; | &#9745; |
| 18 |  GET("/api/public/{api_version}/media/entity") **getListAllEntity()**      | &#9744; | &#9745; |
| 19 |  GET("/api/public/{api_version}/media/entity") **retrieveAnEntity()**    | &#9745; | &#9745; |
| 20 |  GET("/api/public/{api_version}/media/entity/search") **searchEntity()**      | &#9745; | &#9745; |
| 21 |  POST("/api/public/{api_version}/media/entity/playback/token") **getTokenStreaming()**    | &#9745; | &#9745; |
| 22 |  GET("/api/public/v1/cdn/linkplay") **getLinkPlay()**      | &#9745; | &#9744; |
| 23 |  GET("/api/public/v1/cdn/live/linkplay") **getLinkPlayLive()**    | &#9745; | &#9744; |
| 24 |  GET("/api/public/{api_version}/live/entity") (paging) **retrieveALiveEvent()**      | &#9745; | &#9745; |
| 25 |  GET("/api/public/{api_version}/live/entity") **retrieveALiveEvent()**    | &#9745; | &#9745; |
| 26 |  POST("/api/public/{api_version}/live/entity/feed") **startALiveEvent()**      | &#9745; | &#9745; |
| 27 |  GET("/api/public/{api_version}/live/entity/tracking/current-view") **getViewALiveFeed()**    | &#9745; | &#9745; |
| 28 |  GET("/api/public/{api_version}/live/entity/tracking/") **getTimeStartLive()**      | &#9745; | &#9745; |
| 29 |  GET("/api/public/{api_version}/player/info/") **getListSkin()**    | &#9745; | &#9745; |
| 30 |  GET("/api/public/{api_version}/player/info/config/") **getSkinConfig()**      | &#9745; | &#9745; |
| 31 |  GET("/api/public/{api_version}/media/entity/cue-point") **getCuePoint()**    | &#9745; | &#9745; |
| 32 |  GET("/api/public/{api_version}/player/info/config") **getPlayerInfo()**      | &#9745; | &#9744; |
| 33 |  POST("/analytic-tracking/v1/tracking/mobile") **track()**    | &#9745; | &#9745; |
| 34 |  POST("/analytic-tracking/v1/ccu/mobile") **trackCCU()**      | &#9745; | &#9745; |
| 35 |  POST("/analytic-tracking/v2/muiza/eventbulk/mobile") **trackMuiza()**    | &#9745; | &#9745; |
| 36 |  GET("/v1/cdn/ccu/ping") **pingHeartBeat()**      | &#9745; | &#9745; |
| 37 |  GET("http://worldtimeapi.org/api/timezone/Etc/UTC") **getCurrentUTCTime()**    | &#9745; | &#9745; |

**Comment by namnh-0652** [comment]
uizabase
uizacoresdk
uizalivestream

File: uizabase/src/test/java/vn/uiza/restapi/uiza/UZServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+        testSubscriber.assertNotCompleted();
+        testSubscriber.assertError(error);
+        assertEquals(errorBody.getCode(), 400);
+        assertNotNull(errorBody.getData());
+        assertEquals("BAD_REQUEST", errorBody.getType());
+    }
+
+    @Test
+    public void retrieveAnUser_Failed_ServerError() {
+        MockResponse response = new MockResponse().setResponseCode(502);
```
Comment (deathemperor) [inline uizabase/src/test/java/vn/uiza/restapi/uiza/UZServiceTest.java]: we need to make sure all 5xx are handled, not just 502. Should only test for 500, 502, 503, 504.

**Comment by deathemperor** [comment]
@namnh-0652 can you explain why we only test for server error on one API? Is it safe to assume others work the same too?

**Comment by namnh-0652** [comment]
@deathemperor Because the error case depends on the `mockResponse's code` only, not the response body and not the API call function. So, with the `200` code, we can `assert` the response data, with the error cases, we assert that return `HttpException` with expected error code.

**Pull request merged**